internal/common/foraging/deerpopulation.go

```diff
@@ -32,16 +32,16 @@ func createBasicDeerPopulationModel() DeerPopulationModel {
 	}
 	return DeerPopulationModel{
 		deProblem:  simulation.ODEProblem{YPrime: deerPopulationGrowth, Y0: float64(maxDeer), T0: 0, DtStep: 0.1},
-		population: float64(maxDeer),
-		t:          .0,
+		Population: float64(maxDeer),
+		T:          .0,
 	}
 }
 
 // Simulate method simulates the reaction of a deer pop. over i=len(deerConsumption) days where [0, maxDeer] are hunted each day i.
 // Note: if only simulating for one turn ('step'), len(deerConsumption) = 1
 func (dp *DeerPopulationModel) Simulate(deerConsumption []int) {
-	t := dp.t
-	y := dp.population
+	var t float64
```

Comment:
why not dp.T?

Reply:
Got an `ineffectual assignment error` from the `golangci-lint` gods -- that value was never used

Reply:
Is that not a mistake from the gods? I assign `t` to `dp.T` and then update `t` later on so that `dp.T` gets updated implicitly too. Unless this two way coupling doesn't work

Reply:
Oh wait! integers, floats, strings are not reference types, so they are always copied.

I think you meant to do
```
dp.T, dp.Y = deStep(float64(-deerConsumption[i])) // this will update population, t in receiver
```
?

Reply:
This func currently does nothing

Reply:
You can use a pointer to `dp.T` but it's not as elegant. You can, at the end of the function, do

`dp.T, dp.Y = t, y`

The `ineffectual assignment` thing refers to the actual _copied_ values of `dp.t` not being used.

Reply:
(this is one of the pitfalls of object-oriented programming... :( in this case, pointer receiver methods.)